### PR TITLE
dev/core#2329 Add is_active field to Country/StateProvince/County

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Currently it is not possible to disable Country/StateProvince/County elements without deleting them. This make it very hard to update existing CiviCRM sites that are using outdated lists.

This adds the database schema to provide the `is_active` flag - it is not used anywhere yet.

Before
----------------------------------------
No schema to allow disabling Country/StateProvince/County elements.

After
----------------------------------------
Schema is provided.

Technical Details
----------------------------------------
Add is_active flag per other entities.

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
I'm fine with this - it obviously needs an upgrade script - I'll leave if for a few days or possibly put it in the dev digest for visibility / weigh in

**Comment by totten** [comment]
+1 on concept and on need for upgrade script

**Comment by mattwire** [comment]
@eileenmcnaughton @totten Thankyou. Upgrade script now added for 5.35

File: CRM/Upgrade/Incremental/php/FiveThirtyFive.php [hunk 1]
```diff
@@ -52,21 +52,20 @@ public function setPostUpgradeMessage(&$postUpgradeMessage, $rev) {
...
+  public function upgrade_5_35_alpha1(string $rev): void {
+    $this->addTask(ts('Upgrade DB to %1: SQL', [1 => $rev]), 'runSql', $rev);
 
-  // public static function taskFoo(CRM_Queue_TaskContext $ctx, ...) {
-  //   return TRUE;
-  // }
+    $this->addTask('dev/core/#2329 - Add is_active to Country', 'addColumn',
+      'civicrm_country', 'is_active', "tinyint DEFAULT 1 COMMENT 'Is this Country active?'");
+    $this->addTask('dev/core/#2329 - Add is_active to StateProvince', 'addColumn',
+      'civicrm_state_province', 'is_active', "tinyint DEFAULT 1 COMMENT 'Is this StateProvince active?");
```
Comment (seamuslee001) [inline CRM/Upgrade/Incremental/php/FiveThirtyFive.php:66]: @mattwire test fails are because of a missing `'` before the `"` to close the Column Comment

**Comment by mattwire** [comment]
test this please

File: CRM/Upgrade/Incremental/php/FiveThirtyFive.php (hunk 1 above)
Comment (mattwire) [reply CRM/Upgrade/Incremental/php/FiveThirtyFive.php]: Thanks @seamuslee001 now fixed

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
I tested the upgrade & it works. I just want to clarify this before it's merged in terms of whether the existing pseudoconstant code picks up the is_active or not - or what the plan is there

**Comment by mattwire** [comment]
@eileenmcnaughton I guess the next step would be to update pseudoconstant code to filter by is_active by default.

**Comment by eileenmcnaughton** [comment]
I won't hold up merge on the next steps since this is targetting 5.35 which is about to be cut & it will be a pain if it misses it.

However, we will have our normal is_active challenges - ie. only permit active options to be selected when editing, but if an address already has a country that is inactive it should still be viewable & the editing experience is reasonable - e.g let's say I decide to disable 'New Zealand' and put in place 'Te Tai Tokerau'  (succession for Northland!) - when I edit a record with New Zealand I don't want it to be unsavable unless I change it or to be required to change the country to prevent it being blanked

**Pull request merged**